xerrors/xerrors.go

```diff
@@ -16,6 +16,9 @@ import "errors"
 // Calls to [errors.As] and [errors.Is] will be dispatched to the tag first
 // and then fallback to the original error if they don't match the tag.
 func Tag(err, tag error) error {
+	if err == nil {
```

Comment:
won't this hide a programming error?

Reply:
I meant when the user has a call to `xerrors.Tag(err, SomeTagErr)` where `err` is nil by mistake. This way it will return `nil` and "work" instead of panicking later when printing the error (AFAICS it would panic in `err.Error()`).

IMO adding a panic is better because this is a misuse of the API.